processor/transformprocessor/internal/traces/processor.go

```diff
@@ -0,0 +1,88 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type Processor struct {
+	statements []Query
+	logger     *zap.Logger
+}
+
+// Query holds a top level Query for processing trace data. A Query is a combination of a function
+// invocation and the condition to match telemetry for invoking the function.
+type Query struct {
+	function  exprFunc
+	condition condFunc
+}
+
+func NewProcessor(statements []Query, settings component.ProcessorCreateSettings) (*Processor, error) {
+	return &Processor{
+		statements: statements,
+		logger:     settings.Logger,
+	}, nil
+}
+
+func (p *Processor) ProcessTraces(_ context.Context, td pdata.Traces) (pdata.Traces, error) {
+	process(td, p.statements)
+	return td, nil
+}
+
+func process(td pdata.Traces, statements []Query) {
```

Comment:
This can be a func on "Processor" since you pass members anyway, also can be embedded in ProcessTraces.